Source/Boxes.swift

```diff
@@ -35,3 +35,47 @@ extension WeakBox {
         return "WeakBox(\(String(describing: value)))"
     }
 }
+
+/**
+ `ThreadSafeBox` is a helper class that allows use of resource (value) in a thread safe manner.
+ All read and write calls are wrapped in concurrent `DispatchQueue` which protects writing to
+ resource from more than 1 thread at a time.
+ */
+class ThreadSafeBox<T>: CustomDebugStringConvertible {
+    private let queue = DispatchQueue(label: "com.polidea.RxBluetoothKit.ThreadSafeBox", attributes: .concurrent)
```

Comment:
Concurrent queue won't protect you from writing to resource from more then 1 thread.

Reply:
You should switch to serial queue to achieve that.